MULTIARCH#4560: Adding a new API field in images.config.openshift.io (#87753)
openshift/openshift-docs: modules/images-configuration-parameters.adoc

Either `blockedRegistries` or `allowedRegistries` can be set, but not both.

|`imageStreamImportMode`
|Controls the import mode behaviour of image streams. You can set it to either of the following values:

> * `Legacy`: Indicates that the legacy behaviour must be used. The legacy behaviour discards the manifest list and imports a single sub-manifest. In this case, the platform is chosen in the following order of priority: 

Reviewer: Based on https://www.ibm.com/docs/en/ibm-style?topic=grammar-spelling

```suggestion
* `Legacy`: Indicates that the legacy behavior must be used. The legacy behavior discards the manifest list and imports a single sub-manifest. In this case, the platform is chosen in the following order of priority: 
```

> . Tag annotations: Determining the platform by using the platform-specific annotations in the image tags.

Author: You mean an unordered list instead of an ordered list? @Prashanth684 
Since we are defining the priority, I thought an ordered list would be appropriate.
WDYT?

. Control plane architecture or the operating system: Selecting the platform based on the architecture or the operating system of the control plane.

> . `linux/amd64`: If no platform is selected by the above methods, the `linux/amd64` platform is selected. 

Reviewer: Based on https://www.ibm.com/docs/en/ibm-style?topic=word-usage#above (in general, stay away from "directions" that indicate up/down/left/right). 

```suggestion
. `linux/amd64`: If no platform is selected by the preceding methods, the `linux/amd64` platform is selected. 
```

Author: Updated as per the suggestion.

. The first manifest in the list is selected.

> // What uses the behavior? What determines/Selects?

Reviewer 2: When a tag is created in an imagestream, the importmode can be specified as part of the importpolicy : https://github.com/openshift/api/blob/b1a9b9a2e7feee7d6e9fcbc2d13641cd399dbd75/image/v1/types.go#L303 which controls how the image is imported for that tag (single manifest vs manifestlist). The default used to be Legacy. This feature changes that so that the importmode is configured implicitly based on the architecture field in the ClusterVersion's desired status. "multi" would indicate the use of PreserveOriginal as import mode which would import the manifestlist and no entry or an empty architecture field would indicate Legacy which is to import the single manifest : https://github.com/openshift/api/blob/b1a9b9a2e7feee7d6e9fcbc2d13641cd399dbd75/config/v1/types_cluster_version.go#L741

* `PreserveOriginal`: Indicates that the original manifest is preserved. The manifest list and its sub-manifests are imported. 

> If you specify a value for this field, the behavior is applied to the newly created image streams that do not already have a value set. 

Reviewer 2: ```suggestion
If you specify a value for this field, the behavior is applied to the newly created image stream tags that do not already have this value manually set.  Instead of defaulting to `Legacy`, this default will be based on the ClusterVersion's desired status.
```

Author: Regarding this new line "Instead of defaulting to `Legacy`, this default will be based on the ClusterVersion's desired status",
We talk about this behavior in the following para so I think it will be best to exclude this here to avoid redundancy. 
WDYT? @Prashanth684

> // Which value?

Reviewer 2: the importmode value of imagestream tags

Reviewer 2: this i an example you can also provide for the user to visualize where the importmode will be set: https://docs.redhat.com/en/documentation/openshift_container_platform/4.12/html/images/managing-image-streams#importing-manifest-list-through-imagestreamimport_image-streams-managing

Author: Add an xref to the working with manifests section.

> If you do not configure this field, the behaviour is decided based on the payload type advertised by the `ClusterVersion` status. In this case, the platform is chosen as follows:

Reviewer: ```suggestion
If you do not configure this field, the behavior is decided based on the payload type advertised by the `ClusterVersion` status. In this case, the platform is chosen as follows:
```

Either `blockedRegistries` or `allowedRegistries` can be set, but not both.

|`imageStreamImportMode`
|Controls the import mode behavior of image streams.

> You must enable the `TechPreviewNoUpgrade` feature set in the `FeatureGate` custom resource (CR) to enable the `imageStreamImportMode` feature. For more information, see "Understanding feature gates".

Reviewer 3: Can you replace  feature set with featureSet,  and replace feature gates with FeatureGate, other looks to me, thanks

Author: Regarding the suggestion to change feature set to featureSet,
I think feature set is more apt in this case because, featureSet is a field that can be configured in either install-config.yaml file (by using the CLI) or the FeatureGate CR (by using web console).
Also, there are instances of "feature set" in this doc - https://docs.openshift.com/container-platform/4.17/nodes/clusters/nodes-cluster-enabling-features.html#nodes-cluster-enabling-features-install_nodes-cluster-enabling

So I suggest keeping this as is.
WDYT? @wewang58

> You must enable the `TechPreviewNoUpgrade` feature set in the `FeatureGate` custom resource (CR) to enable the `imageStreamImportMode` feature. For more information, see "Understanding feature gates".

Reviewer 4: I do not see a link for Understanding feature gates in the document. COuld such a link be added to an additional resources section?

Reviewer 4: OK, an ifndef statement is needed here for Dedicated and ROSA docs as that link does not apply?

Author: That's correct! I've excluded this field from the ROSA and OSD docs as confirmed by the SME @Prashanth684
